packages/teams-js/src/private/copilot/sidePanelInterfaces.ts

```diff
@@ -0,0 +1,332 @@
+  calendarInvites?: CalendarInviteContent[];
+  webPages?: WebPageContent[];
+  otherContent?: Array<Record<string, unknown>> | undefined; // Other content types that don't fit into the above categories
+}
+
+/**
+ * @hidden
+ *
+ * Interface for content data we get from hub
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export type ContentItem =
+  | EmailContent
+  | TextSelection
+  | MediaSelection
+  | CalendarInviteContent
+  | WebPageContent
+  | MixedContent;
+
+export interface Content {
+  userAction?: string;
+  contentType:
+    | ContentItemType.CALENDAR_INVITE
+    | ContentItemType.EMAIL
+    | ContentItemType.MEDIA
+    | ContentItemType.TEXT
+    | ContentItemType.WEB_PAGE
+    | ContentItemType.MIXED;
+  formCode?: string; // Unique identifier for the content
+  contentItems: ContentItem[];
+  metadata?: string;
+  description?: string;
+  error_code?: string; // Optional error code if the content retrieval failed
+  status?: string; // Optional status message
+}
+
+/**
+ * @hidden
+ *
+ * Interface for the response context used during user consent pre-checks.
+ * Contains information about the user's consent status and whether to show the consent card.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export interface PreCheckContextResponse {
+  error_code?: string; // Optional error code if the pre-check failed
+  status?: string; // Optional status message
+  user_consent: UserConsent; // The user's consent status (accepted or not accepted)
+  show_consent_card: boolean; // Whether the consent card should be shown to the user
+}
+
+/**
+ * @hidden
+ *
+ * Enum representing possible user consent states.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export enum UserConsent {
+  Accepted = 'accepted', // User has accepted/consented
+  NotAccepted = 'not_accepted', // User has not accepted/consented
+}
+
+/**
+ * @hidden
+ *
+ * Type for user action handler functions that receive content data.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export enum SidePanelErrorCode {
+  BLOCKED_BY_POLICY = 'page_content_blocked_by_policy',
+  BLOCKED_BY_DLP = 'page_content_blocked_by_dlp',
+  MEDIA_NOT_SUPPORTED = 'media_not_supported',
+  USER_CONSENT_REQUIRED = 'user_consent_required',
+  EXTRACTION_FAILED = 'content_extraction_failed',
+  CONTENT_NOT_FOUND = 'content_not_found',
+  CONTENT_NOT_SUPPORTED = 'content_not_supported',
+  CONTENT_CHANGED = 'content_changed',
+  UNKNOWN_ERROR = 'unknown_error',
+  INTERNAL_ERROR = 'internal_error', // catch all error code for unexpected issues
+  NOT_SUPPORTED_ON_PLATFORM = 'not_supported_on_platform', // API not supported on the current platform
+}
+
+/**
+ * @hidden
+ *
+ * Interface for errors related to side panel operations.
+ * Contains an error code and an optional message.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export interface SidePanelError {
+  errorCode: SidePanelErrorCode | ErrorCode;
```

Comment:
This menas errorCode can be numbers like: 
```
  NOT_SUPPORTED_ON_PLATFORM = 100,
  INTERNAL_ERROR = 500,
  NOT_SUPPORTED_IN_CURRENT_CONTEXT = 501,
```
OR the strings like:
```
  BLOCKED_BY_POLICY = 'page_content_blocked_by_policy',
  BLOCKED_BY_DLP = 'page_content_blocked_by_dlp',
  MEDIA_NOT_SUPPORTED = 'media_not_supported',
```
I also left another comment, but I would vote for keeping it as `SdkError` type to match with other capabilities. If we have to have `SidePanelErrorCode`, I would keep it to `SidePanelError` type instead of combining both.